disperser/dataapi/server.go

```diff
@@ -333,10 +335,105 @@ func (s *server) FetchBlobHandler(c *gin.Context) {
 	}
 
 	s.metrics.IncrementSuccessfulRequestNum("FetchBlob")
-	c.Writer.Header().Set(cacheControlParam, fmt.Sprintf("max-age=%d", maxFeedBlobAage))
+	c.Writer.Header().Set(cacheControlParam, fmt.Sprintf("max-age=%d", maxFeedBlobAge))
 	c.JSON(http.StatusOK, metadata)
 }
 
+// FetchBlobsFromBatchHeaderHash godoc
+//
+//	@Summary	Fetch blob metadata by batch header hash
+//	@Tags		Feed
+//	@Produce	json
+//	@Param		batch_header_hash	path		string	true	"Batch Header Hash"
+//	@Param		limit				query		int		false	"Limit [default: 10]"
+//	@Param		next_token			query		string	false	"Next page token"
+//	@Success	200					{object}	BlobsResponse
+//	@Failure	400					{object}	ErrorResponse	"error: Bad request"
+//	@Failure	404					{object}	ErrorResponse	"error: Not found"
+//	@Failure	500					{object}	ErrorResponse	"error: Server error"
+//	@Router		/feed/batches/{batch_header_hash}/blobs [get]
+func (s *server) FetchBlobsFromBatchHeaderHash(c *gin.Context) {
+	timer := prometheus.NewTimer(prometheus.ObserverFunc(func(f float64) {
+		s.metrics.ObserveLatency("FetchBlobsFromBatchHeaderHash", f*1000) // make milliseconds
+	}))
+	defer timer.ObserveDuration()
+
+	batchHeaderHash := c.Param("batch_header_hash")
+
+	limit, err := strconv.Atoi(c.DefaultQuery("limit", "100"))
+	if err != nil || limit <= 0 {
```

Comment:
shouldn't we return if `err != nil` as it means the request is not valid?

Reply:
Was following what we do for `/feed/blobs` which is pretty much just being lenient for the limit param. I'll go ahead and change it though.